### PR TITLE
[Enhance] support spatial-temporal detection demo

### PR DESCRIPTION
**Pull request by hukkai**
Can be reviewed, but do not merge util `mmdet` supporting `FastRCNN` model in the 3.x branch.

### CODE REVIEW

File: demo/demo_spatiotemporal_det.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+                    diag0 = (location[0] + textwidth, location[1] - 14)
+                    diag1 = (location[0], location[1] + 2)
+                    cv2.rectangle(frame, diag0, diag1, plate[k + 1], -1)
+                    cv2.putText(frame, text, location, FONTFACE, FONTSCALE,
+                                FONTCOLOR, THICKNESS, LINETYPE)
+
+    return frames_
+
+
+def frame_extraction(video_path):
```
Comment (Dai-Wenxun) [inline demo/demo_spatiotemporal_det.py:105]: please use the api `frame_extract` in `mmaction.utils.misc`.

**Review by Dai-Wenxun — COMMENTED** [review]
add related doc

File: mmaction/apis/inference.py [hunk 2]
```diff
@@ -126,8 +126,8 @@ def detection_inference(det_config: Union[str, Path, mmengine.Config],
         raise ImportError('Failed to import `inference_detector` and '
                           '`init_detector` from `mmdet.apis`. These apis are '
                           'required in this inference api! ')
-
-    model = init_detector(det_config, det_checkpoint, device)
+    model = init_detector(
```
Comment (Dai-Wenxun) [inline mmaction/apis/inference.py:129]: Why did you change this line?

**Comment by hukkai** [comment]
Will open a new PR

**Comment by hukkai** [comment]
Closed and re-opened in https://github.com/open-mmlab/mmaction2/pull/2019
